### PR TITLE
TC05 - System shows message when user enters wrong password several times

### PR DESCRIPTION
**Pull request by sattdn21-01-02**

### CODE REVIEW

File: src/main/java/helper/Constant.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
     public static final int TIMES_WAIT_ELEMENTS = 120;
     public static final int LONG_TIME_WAIT = 30;
     public static final String DATE_FORMAT = "M/d/yyyy";
-    public static final String FAIL_MSG_LOGIN_MULTIPLE = "You have used 4 out of 5 login attempts. After all 5 have been used, you will be unable to login for 15 minutes.";
+    public static final String INVALID_LOGIN_MULTIPLE_MSG = "You have used 4 out of 5 login attempts. After all 5 have been used, you will be unable to login for 15 minutes.";
```
Comment (thanhlelgg) [inline src/main/java/helper/Constant.java:14]: should be: 
MAXIUM_LOGIN_ATTEMPTS_WARNING_MSG

File: src/test/java/com/logigear/LoginTest.java [hunk 2]
```diff
@@ -9,13 +18,31 @@
...
+        LoggerHelper.startTestCase("TC05 - System shows message when user enters wrong password several times ");
+
+        homePage.goToLoginPage();
+
+        account.setPassword(DataHelper.getRandomErrorPassword());
+        loginPage.login(account);
+
+        String actualMsg = loginPage.getGeneralErrorMessage();
+        String expectedMsg = Constant.INVALID_LOGIN_MULTIPLE_MSG;
+        Assert.assertEquals(actualMsg, expectedMsg);
```
Comment (fey031) [inline src/test/java/com/logigear/LoginTest.java:42]: missing assert message

File: src/test/java/com/logigear/LoginTest.java [hunk 3]
```diff
@@ -9,13 +18,31 @@
...
+    public void TC05(Account account) {
+        LoggerHelper.startTestCase("TC05 - System shows message when user enters wrong password several times ");
+
+        homePage.goToLoginPage();
+
+        account.setPassword(DataHelper.getRandomErrorPassword());
+        loginPage.login(account);
+
+        String actualMsg = loginPage.getGeneralErrorMessage();
+        String expectedMsg = Constant.INVALID_LOGIN_MULTIPLE_MSG;
```
Comment (fey031) [inline src/test/java/com/logigear/LoginTest.java:41]: if you define this message in Constant, don't need declare new variable for it

File: src/main/java/helper/Constant.java (hunk 1 above)
Comment (sattdn21-01-02) [reply src/main/java/helper/Constant.java:14]: Thanks for your review. I just have fixed and pushed into this branch. Please review my newest update.

**Review by fey031 — CHANGES_REQUESTED** [review]

File: src/test/java/com/logigear/LoginTest.java (hunk 2 above)
Comment (sattdn21-01-02) [reply src/test/java/com/logigear/LoginTest.java:42]: Thanks for your review. I just have fixed and pushed into this branch. Please review my newest update.

File: src/test/resources/test-data/login-data.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    {
+      "email": "tnminh19971234@gmail.com",
+      "password": "minhvip1997"
+    },
+    {
+      "email": "tnminh19973215@gmail.com",
+      "password": "doubleminh1997"
+    }
+  ],
+  "account_error": [
```
Comment (tanmle) [inline src/test/resources/test-data/login-data.json:12]: what is difference between account_error and account_invalid?

File: src/test/resources/test-data/login-data.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+  "accounts": [
```
Comment (tanmle) [inline src/test/resources/test-data/login-data.json:2]: where do you need this?

**Review by tanmle — CHANGES_REQUESTED** [review]

**Review by tanmle — CHANGES_REQUESTED** [review]
you don't understand the test cases

File: src/main/java/page_objects/LoginPage.java [hunk 6]
```diff
@@ -25,6 +25,15 @@ public void login(Account account) {
         this.btnLogin.click();
     }
 
+    public void loginNTimes(Account account, int n) {
+        for(int i=0;i<n;i++){
```
Comment (tanmle) [inline src/main/java/page_objects/LoginPage.java:29]: format code.

File: src/main/java/page_objects/LoginPage.java [hunk 7]
```diff
@@ -25,6 +25,15 @@ public void login(Account account) {
...
+            this.txtEmail.enterText(account.getEmail());
```
Comment (tanmle) [inline src/main/java/page_objects/LoginPage.java:30]: DRY.

File: src/main/java/page_objects/LoginPage.java (hunk 6 above)
Comment (thanhlelgg) [inline src/main/java/page_objects/LoginPage.java:29]: spacing

File: src/main/java/page_objects/LoginPage.java (hunk 7 above)
Comment (thanhlelgg) [inline src/main/java/page_objects/LoginPage.java:30]: you already have a login function right? why don't you just call it here?

File: src/test/java/com/logigear/LoginTest.java [hunk 8]
```diff
@@ -10,11 +19,17 @@
 import page_objects.HomePage;
 import page_objects.LoginPage;
 
+import java.io.FileReader;
```
Comment (thanhlelgg) [inline src/test/java/com/logigear/LoginTest.java:22]: what do you need this for? doesn't look like you are reading anything?

File: src/main/java/page_objects/LoginPage.java (hunk 6 above)
Comment (sattdn21-01-02) [reply src/main/java/page_objects/LoginPage.java:29]: Thank you for review , i will format code now

File: src/test/java/com/logigear/LoginTest.java [hunk 9]
```diff
@@ -10,11 +19,17 @@
...
+import java.io.IOException;
```
Comment (thanhlelgg) [inline src/test/java/com/logigear/LoginTest.java:23]: this one is not used as well

**Review by thanhlelgg — CHANGES_REQUESTED** [review]

File: src/main/java/page_objects/LoginPage.java (hunk 7 above)
Comment (sattdn21-01-02) [reply src/main/java/page_objects/LoginPage.java]: Sorry, i will call function login to replace code

File: src/test/java/com/logigear/LoginTest.java (hunk 8 above)
Comment (sattdn21-01-02) [reply src/test/java/com/logigear/LoginTest.java:22]: Sorry , I will delete all import unnecessary in the last update

File: src/main/java/page_objects/LoginPage.java (hunk 7 above)
Comment (sattdn21-01-02) [reply src/main/java/page_objects/LoginPage.java]: thank you for review, I will change the code in the last update

File: src/main/java/page_objects/LoginPage.java (hunk 6 above)
Comment (sattdn21-01-02) [reply src/main/java/page_objects/LoginPage.java]: Thank you for review, i Iwill format the code now

**Review by thanhlelgg — APPROVED** [review]

**Pull request merged**